Userland/cat.cpp

```diff
@@ -25,27 +25,55 @@
  */
 
 #include <AK/Vector.h>
+#include <LibCore/ArgsParser.h>
 #include <assert.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <sys/stat.h>
 #include <string.h>
 #include <unistd.h>
 
+bool detect_useless_use_of_cat(int argc)
+{
+    // I can think of somewhat valid uses of zero-argument cat
+    // from a tty to a pipe, so let's allow those.
+    if (argc != 2)
+        return false;
+
+    struct stat statbuf;
+    int rc = fstat(1, &statbuf);
+    if (rc < 0)
+        return false;
+
+    return S_ISFIFO(statbuf.st_mode);
+}
+
 int main(int argc, char** argv)
 {
     if (pledge("stdio rpath", nullptr) < 0) {
         perror("pledge");
         return 1;
     }
 
+    Vector<const char*> paths;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(paths, "File path", "path", Core::ArgsParser::Required::No);
```

Comment:
File path _s_?

Reply:
That's questionable — it looks like this, for some context:
![image](https://user-images.githubusercontent.com/10091584/84831085-630f7700-b033-11ea-9c1f-f503fda61e79.png)
and I would say `path` is *one* path which can be repeated (but most of the time it's used with a single path). This is also what other users of `Core::ArgsParser` seem to do in similar cases.

Reply:
Oh, it puts dots after it.
My bad. 'path' makes more sense.